sql27thsep/sql27th sep.txt

```diff
@@ -0,0 +1,224 @@
+
+
+10. Write a query to find the byte count of an empty string and explain the result.
+select datalength('');_________________________output=0 bcoz here nothing is to be stored 
+
+11. Find the position of 'Server' in the string 'Microsoft SQL Server'.
+select CHARINDEX('server','microsoft sql server');
+
+12. From an 'Emails' table, write a query to extract the domain name from email addresses.
+select substring(emailid,charindex('@',emailid),len(emailid)) from employees;
+
+13. Find the position of the first number in the string 'ABC123DEF456'.
+select PATINDEX('%[0-9]%','ABC123DEF456');
+
+14. Write a query to find all product names from a 'Products' table that contain a number.
+select productname from products where patindex('%[0-9]%',productname)>0;
+
+15. Join the strings 'SQL', 'Server', and '2022' with spaces between them.
+select CONCAT_WS('-','sql','server','2022');
+
+16. From 'Employees' table with 'FirstName' and 'LastName' columns, create a 'FullName' column.
+select CONCAT_WS(' ',firstname,lastname) as fullname from employees;
+
+17. Join the array ('SQL', 'Server', '2022') with a hyphen as the separator.
+select CONCAT_WS('-','sql','server','2022');
+
+18. From an 'Addresses' table, combine 'Street', 'City', 'State', and 'ZIP' columns into a single address string.
+select concat_ws(',',street,city,state,zip) from address;
+
+19. Change all occurrences of 'a' to 'e' in the string 'database management'.
+select replace('database management','a','e');
+
+20. From a 'Products' table, write a query to replace all spaces in product names with underscores.
+select replace(productname,' ','-') from products;
+
+21. Create a string of 10 asterisks (*).
+select replicate('*',10);
+
+22. Write a query to pad all product codes in a 'Products' table to a length of 10 characters with leading zeros.
+select right(replicate('0',10)+[product code],10) from products;
+
+23. Insert the string 'New ' at the beginning of 'York City'.
+select concat_ws(' ','new'+ 'York city');    
+
+24. From an 'Emails' table, mask the username part of email addresses, showing only the first and last characters.
+select concat_ws('',left(emailID,1),replicate('*',charindex('@',emailID)),right(emailID,len(emailID)-charindex('@',emailID)+2)) from employees;
+
+25. Convert the string 'sql server' to uppercase.
+select upper('sql server');
+
+26. Write a query to convert all customer names in a 'Customers' table to uppercase.
+select upper(customername) from customers;
+
+27. Convert the string 'SQL SERVER' to lowercase.
+select lower('SQL SERVER');
+
+28. From a 'Products' table, write a query to convert all product descriptions to lowercase.
+select lower(productdescription) from products;
+
+29. Remove trailing spaces from the string 'SQL Server    '.
+select rtrim('SQL Server    ', ' ');
+
+30. Write a query to remove trailing spaces from all email addresses in an 'Employees' table.
+select rtrim(emailID,' ') from employees;
```

Comment:
we don't need to pass the space , right ?

Reply:
wasnt aware about that it works like one argument to remove spaces,now got it. yeah then there was no need to add space as 2nd argument

Reply:
do check about TRIM function also.